pandas/core/common.py

```diff
@@ -430,3 +431,46 @@ def _pipe(obj, func, *args, **kwargs):
         return func(*args, **kwargs)
     else:
         return func(obj, *args, **kwargs)
+
+
+def searchsorted_integer(arr, value, side="left", sorter=None):
+    dtype = arr.dtype
+    if sorter is not None:
+        sorter = ensure_platform_int(sorter)
+
+    # check integer bounds
+    iinfo = np.iinfo(dtype)
+    value_arr = np.array([value]) if is_scalar(value) else np.array(value)
+    if (value_arr < iinfo.min).any() or (value_arr > iinfo.max).any():
+        msg = "Value {} out of bound for dtype {}".format(value, dtype)
+        raise ValueError(msg)
```

Comment:
Notice I raise here when value is out of bound. I think this is reasonable behaviour and avoids some overflow surprises.